src/pages/index.js

```diff
@@ -23,59 +79,98 @@ const userInfo = new UserInfo({
+    .then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+      });
+      editProfilePopup.close();
+    })
+    .catch(console.error)
+    .finally(() => editProfilePopup.setButtonText("Save"));
 });
 editProfilePopup.setEventListeners();
 
-// Updated Add Card Popup ✅
 const addCardPopup = new PopupWithForm(addCardModal, (data) => {
-  console.log("Form Data (Add Card): ", data); // Debugging
-
-  const cardElement = createCard({
-    name: data["title"] || "Untitled", // Uses name="title" from HTML
-    link: data["image url"] || "https://via.placeholder.com/150", // Uses name="image url" from HTML
-  });
-  cardSection.addItem(cardElement);
-  addCardPopup.close();
-  addCardForm.reset(); // reset -- Thank you!!!
-  addCardValidator.toggleButtonState(); // this will disable the button --- Thank You!!!
+  addCardPopup.setButtonText("Saving...");
+  api
+    .addCard({ name: data["title"], link: data["image url"] })
+    .then((cardData) => {
+      const cardElement = createCard(cardData);
+      cardSection.addItem(cardElement);
+      addCardForm.reset();
+      addCardValidator.toggleButtonState();
+      addCardPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => addCardPopup.setButtonText("Save"));
 });
 addCardPopup.setEventListeners();
 
+const editAvatarPopup = new PopupWithForm("#edit-avatar-modal", (data) => {
+  editAvatarPopup.setButtonText("Saving...");
+  api
+    .updateUserAvatar(data.avatar)
+    .then((userData) => {
+      profileImage.src = userData.avatar;
+      editAvatarPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => editAvatarPopup.setButtonText("Save"));
+});
+editAvatarPopup.setEventListeners();
+
 // PopupWithImage Instance
 const previewPopup = new PopupWithImage(cardPreviewModal);
 previewPopup.setEventListeners();
 
-// Create Card Function ✅
+// Create Card Function
 function createCard(data) {
-  const card = new Card(data, "#card-template", (name, link) => {
-    previewPopup.open({ name, link });
-  });
-  return card.generateCard();
+  const card = new Card(
+    data,
+    "#card-template",
+    (name, link) => previewPopup.open({ name, link }),
+    (cardElement) => {
+      cardToDelete = cardElement;
+      openModal(confirmModal);
+    }
+  );
+  const element = card.generateCard();
+  element.setAttribute("data-id", data._id);
+  return element;
 }
 
-// Section Instance - Initial Cards ✅
+// Section Instance
 const cardSection = new Section(
   {
-    items: initialCards,
     renderer: (item) => {
       const cardElement = createCard(item);
       cardSection.addItem(cardElement);
     },
   },
   cardListElement
 );
-cardSection.renderItems();
+
+// Load user data and cards from API
+api
+  .getAppData()
+  .then(([userData, cards]) => {
+    userInfo.setUserInfo({ name: userData.name, description: userData.about });
+    profileImage.src = userData.avatar;
```

Comment:
`      profileImage.src = userData.avatar;
`

Only methods of the `UserInfo` class should set/get the profile data (including the **avatar**)